common/middleware/instrument.go

```diff
@@ -40,20 +40,30 @@ func (i Instrument) Wrap(next http.Handler) http.Handler {
 	})
 }
 
-var invalidChars = regexp.MustCompile(`[^a-zA-Z0-9]+`)
-
+// Return a name identifier for ths request.  There are three options:
+//   1. The request matches a gorilla mux route, with a name.  Use that.
+//   2. The request matches an unamed gorilla mux router.  Munge the path
+//      template such that templates like '/api/{org}/foo' come out as
+//      'api_org_foo'.
+//   3. The request doesn't match a mux route.  Munge the Path in the same
+//      manner as (2).
+// We do all this as we do not wish to emit high cardinality labels to
+// prometheus.
```

Comment:
If MakeLabelValue will work for all the routes, can we remove options 1 and 2 (including removing named routes and all that supporting code) and just use 3?